### DISCUSSION TITLE
Using standard npm registry search api

### DISCUSSION THREAD

**Discussion by kizivat** [verdaccio/verdaccio#2902] [Q&A] [answered]
The issue #310 fixed by PR #1732 suggests that the endpoint `/-/v1/search` should work as is specified here: https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#get-v1search

However I only get 404 response from that endpoint. Do I need to somehow turn this feature on? I'm unable to find anything on this in the docs.

**Comment by juanpicado** [comment]
If you specify which version you are using that would help, should works out of the box.

**Reply by kizivat** [reply]
At the moment I'm using 5.4.0, but I the 404 error I was getting was at 4.2.

It works in 5.4.0 when I `curl` the endpoint, however I my requests get rejected when I use REST client (Thunder Client) or JS `fetch` API. That's likely different issue though.

**Reply by juanpicado** [reply]
What about npm search command ? I'd suggest reproducible steps because could be anything.

**Comment by kizivat** [comment]
`npm search` does work as well

```sh
$ npm search xyz --registry=http://localhost:4873/
> No matches found for "xyz"
```

this is OK.

I'm not sure though, why I get error when I do `GET http://localhost:4873/-/v1/search?text=xyz` via [Thunder Client VS Code extension](https://marketplace.visualstudio.com/items?itemName=rangav.vscode-thunder-client).

**Reply by juanpicado** [reply]
Could you share what you see in the server log? I'm blind now.

**Reply by kizivat** [reply]
Hmm I set the log level to `trace` and looks like the request doesn't event hit the server (nothing gets logged at all.

I get feeling it has something to do with the fact that the request is sent from within VS Code. Neither do other REST client extensions work with Verdaccio's registry API.

The most sensible error message I got from one of the extensions is


> The connection was rejected. Either the requested service isn’t running on the requested server/port, the proxy settings in vscode are misconfigured, or a firewall is blocking requests. Details: RequestError: connect ECONNREFUSED 127.0.0.1:4873.


I tried running something else on the same `localhost:4873` and the request have made it through (and visa versa - verdaccio listening on other port => requests still didn't go through).

I don't use proxy

**Comment by kizivat** [comment, marked as answer]
I managed to make it work when I used my PC's local IP instead of `localhost` and set:

```yml
listen:
  - http://{local_ip}:4873
```

in Vedaccio config
